numpy/core/tests/test_datetime.py

```diff
@@ -471,6 +471,19 @@ def test_timedelta_nat_format(self):
         # gh-17552
         assert_equal('NaT', '{0}'.format(np.timedelta64('nat')))
 
+    def test_format(self):
+        vs = "2021-12-12T10:10:10"
+        v = np.datetime64(vs)
+        with assert_raises_regex(
+            ValueError,
+            "Unknown format code"
+        ):
+            format(v, "+.2f")
+
+        # TODO: this behaviour should not occur
```

Comment:
Just a note: This is not behavior that was changed and the array code path rejects the format spec for non 0-D arrays (we drop through for 0-D).